lib/Sema/TypeCheckMacros.cpp

```diff
@@ -759,19 +767,23 @@ bool swift::expandFreestandingDeclarationMacro(
   PrettyStackTraceDecl debugStack(
       "type checking expanded declaration macro", med);
 
-  // Retrieve the parsed declarations from the list of top-level items.
   auto topLevelItems = macroSourceFile->getTopLevelItems();
-  for (auto item : topLevelItems) {
-    auto *decl = item.dyn_cast<Decl *>();
-    if (!decl) {
-      ctx.Diags.diagnose(
-          macroBufferRange.getStart(), diag::expected_macro_expansion_decls);
-      return false;
-    }
-    decl->setDeclContext(dc);
-    results.push_back(decl);
+  // Set the decl context of decls.
+  for (auto node : topLevelItems)
+    if (auto *decl = node.dyn_cast<Decl *>())
+      decl->setDeclContext(dc);
+
+  // When top-level, wrap every expression in a `TopLevelCodeDecl`.
+  if (dc->getContextKind() == DeclContextKind::FileUnit) {
+    SmallVector<ASTNode, 4> expansion(topLevelItems.begin(),
+                                      topLevelItems.end());
+    for (auto &node : expansion)
+      if (auto *expr = node.dyn_cast<Expr *>())
```

Comment:
We should handle `Stmt *` here as well, right?

Reply:
I'll handle this in the next patch when I added support for codeItem.